job_executor/model/metadata.py

```diff
@@ -255,10 +253,54 @@ def patch(self, other: 'Variable'):
             patched.update({"format": self.format})
         if self.key_type is not None:
             patched.update({
-                'keyType': self.key_type.patch(other.key_type).dict()
+                'keyType': self.key_type.dict() if only_patch_description else
+                self.key_type.patch(other.key_type).dict()
             })
         return Variable(**patched)
 
+    def validate_patching_fields(
+        self, other, with_name: bool = False, with_key_type: bool = False
+    ):
+        caption = 'Illegal change to one of these variable fields: \n'
+        message = ''
+        if (
+            self.data_type != other.data_type or
+            self.format != other.format or
+            self.variable_role != other.variable_role
+        ):
+            message = (
+                f'dataType: {self.data_type} to {other.data_type},'
+                f'format: {self.format} to {other.format},'
+                f'variable_role: {self.variable_role} to '
+                f'{other.variable_role}'
+            )
+        if (
+            with_name and
+            self.name != other.name
+        ):
+            message += f'name: {self.name} to {other.name},'
+        if (
+            with_key_type and
+            self.key_type != other.key_type
+        ):
+            message += f'name: {self.key_type} to {other.key_type},'
+
+        if message:
+            raise PatchingError(caption + message)
+
+    def validate_patching_for_all_variable_roles(self, other: 'Variable'):
+        if self.key_type is None and other.key_type is not None:
+            raise PatchingError('Can not change keyType')
+        if len(self.represented_variables) != len(other.represented_variables):
```

Comment:
This line would likely fail a dataset with KOMMUNE in measure where the kommune codelist has been centrally updated. As "code list periods" => represented variables

Reply:
True, I fix!